client/src/components/AvailableDeskList.tsx

```diff
@@ -0,0 +1,63 @@
+import { List } from "@mui/material";
+import { useState, FC, useEffect, useContext } from "react";
+import { DeskType, Booking } from "../types";
+import { DeskItem } from "./DeskItemButton";
+import { FormContext, getUnixTime } from "../pages/bookingPage";
+import { isBetween } from "../helperFunctions";
+
+export const AvailableDeskList: FC = () => {
+  const [currentDeskList, setCurrentDeskList] = useState([] as DeskType[]);
+  const {
+    startTime: [startTimeValue],
+  } = useContext(FormContext);
+  const {
+    endTime: [endTimeValue],
+  } = useContext(FormContext);
+  const {
+    date: [dateValue],
+  } = useContext(FormContext);
+
+  useEffect(() => {
+    const setDeskList = async () => {
+      const bookingList = await fetch(
+        `${process.env.REACT_APP_ROOT_URL}api/bookinglist`
+      );
+      const jsonBookingList = await bookingList.json();
+      const bookings: Booking[] = jsonBookingList.bookingList;
+
+      const bookingsAtTime = bookings.filter((booking: Booking) => {
+        const unixStartTime = getUnixTime(dateValue, startTimeValue);
+        const unixEndTime = getUnixTime(dateValue, endTimeValue);
+        if (
```

Comment:
you could return the boolean directly in declarative style. This is a matter of taste though:

```
const isBetweenStart = isBetween(booking.start_time, unixStartTime, unixEndTime);
const isBetweenEnd = isBetween(booking.end_time, unixStartTime, unixEndTime);
return isBetweenStart || isBetweenEnd;
```